DatadogSessionReplay/SRExample/SRSnapshotTests/Utils/ImageComparison.swift

```diff
@@ -0,0 +1,249 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import UIKit
+import XCTest
+
+internal struct ReferenceImage {
+    let url: URL
+
+    private init(path: String, file: StaticString = #filePath) {
+        self.url = URL(fileURLWithPath: "\(file)", isDirectory: false)
+            .deletingLastPathComponent()
+            .appendingPathComponent(path)
+    }
+
+    /// Creates reference image in folder with given name.
+    /// The folder will be placed next to current file.
+    /// The image will be named by the name of current test and suffixed with `imageFileSuffix`.
+    static func inFolder(
+        named folderName: String,
+        imageFileSuffix: String = "",
+        file: StaticString = #filePath,
+        function: StaticString = #function
+    ) -> ReferenceImage {
+        return ReferenceImage(path: "\(folderName)/\(function)\(imageFileSuffix).png", file: file)
+    }
+}
+
+/// Compares image against reference file OR updates reference file with image data (if `record == true`).
+///
+/// It raises `XCTest` assertion failure if image is different than reference.
+///
+/// - Parameters:
+///   - image: the image to compare OR record
+///   - referenceImage: the reference file to compare against
+///   - record: if `true`, then reference file will be created / overwritten with `image` data
+///   - file: `#filePath` for eventual `XCTest` assertion failure
+///   - line: `#line` for eventual `XCTest` assertion failure
+internal func compare(
+    image: UIImage,
+    referenceImage: ReferenceImage,
+    record: Bool,
+    file: StaticString = #filePath,
+    line: UInt = #line
+) throws {
```

Comment:
I think we should split this in 3 functions:
- Assert Device
- Record Image
- Compare Image

Also, I think we should leverage the `DDAssert` pattern for all assertion. For that, we would need to make `_DDEvaluateAssertion` and `DDAssertError` public, this way all module could use it to define their assertion. WDYT?

Reply:
I like these ideas 👍 , let me look into this it this PR 👀